### PR TITLE
Refactor function handler

### PR DESCRIPTION
**Pull request by maxcapodi78**

### CODE REVIEW

File: .github/workflows/unit_tests.yml [hunk 1]
```diff
@@ -132,13 +132,13 @@ jobs:
...
         with:
-          max_attempts: 3
+          max_attempts: 2
           retry_on: error
           timeout_minutes: 50
           command: |
             testenv\Scripts\Activate.ps1
             Set-Item -Path env:PYTHONMALLOC -Value "malloc"
-            pytest -n 6 --dist loadfile --durations=50 -v --cov=pyaedt --cov-report=xml --cov-report=html --junitxml=junit/test-results.xml _unittest
+            pytest -n 4 --dist loadfile --durations=50 -v --cov=pyaedt --cov-report=xml --cov-report=html --junitxml=junit/test-results.xml _unittest
```
Comment (SMoraisAnsys) [inline .github/workflows/unit_tests.yml:141]:
```suggestion
            pytest -n auto --dist loadfile --durations=50 -v --cov=pyaedt --cov-report=xml --cov-report=html --junitxml=junit/test-results.xml _unittest
```

**Review by SMoraisAnsys — CHANGES_REQUESTED** [review]
Since the line of codes
```
            if not settings.enable_error_handler:
                if settings.release_on_exception:
                    from pyaedt.generic.desktop_sessions import _desktop_sessions

                    for v in list(_desktop_sessions.values())[:]:
                        v.release_desktop(v.launched_by_pyaedt, v.launched_by_pyaedt)
                raise e
            else:
                return False
```
are used in every exception handling, could we have an inner function in `wrapper` to refactor that and call it every time it is needed ?

File: .github/workflows/unit_tests.yml (hunk 1 above)
Comment (maxcapodi78) [reply .github/workflows/unit_tests.yml:141]: here the problem is that is not related to the number of cores available but on the issues caused by running parallel AEDT sessions. I'm trying to limit to make the test more stable and reduce randomic failures

**Comment by maxcapodi78** [comment]
> Since the line of codes
> 
> ```
>             if not settings.enable_error_handler:
>                 if settings.release_on_exception:
>                     from pyaedt.generic.desktop_sessions import _desktop_sessions
> 
>                     for v in list(_desktop_sessions.values())[:]:
>                         v.release_desktop(v.launched_by_pyaedt, v.launched_by_pyaedt)
>                 raise e
>             else:
>                 return False
> ```
> 
> are used in every exception handling, could we have an inner function in `wrapper` to refactor that and call it every time it is needed ?

Implemented

**Review by SMoraisAnsys — APPROVED** [review]
LGTM

**Review by Samuelopez-ansys — APPROVED** [review]

**Review by SMoraisAnsys — APPROVED** [review]

**Pull request merged**